osbs/build/build_request.py

```diff
@@ -377,17 +377,34 @@ def adjust_for_registry_api_versions(self):
     def adjust_for_triggers(self):
         """
         Remove trigger-related plugins if no triggers set
+        import_image plugin is only removed for non-custom base images
```

Comment:
Whether or not triggers are used is defined per environment, not per image. 
When autorebuild is enabled in a environment, all images will have triggers in their BuildConfig - even images that have opted out of autorebuild. 
The only exception is custom base images - since no other images trigger it, the triggers are always removed to avoid incorrectly tracking `koji/image-build`.